bert_e/docs/USER_DOC.md

```diff
@@ -471,8 +471,76 @@ to progress to the next step.  message code
 
 Queues
 ------
-A mechanism to stack pull requests and reduced the number of builds has been
-implemented in Bert-E 2.0. This is not documented here yet.
+
+Bert-E's queue is a mechanism to stack and merge pull requests
+that have passed all the [conditions](#conditions-to-merge-a-pull-request).
+
+When multiple people are working on the same repository, there
+will be multiple pull requests opened, each of them at
+different stages of meeting the conditions set by the bot.
+Therefore when the conditions are met, they will not contain
+the latest commits that are included in the destination branch.
+
+Bert-E will proceed in taking the pull request forward by merging it into
+the queue branches.
+
+Each development/x.y branch will have it's according `q/x.y` branch, and
+each pr in the queue will also have it's according `q/$PR_ID/x.y/$BRANCH_NAME`
+
+For a PR that is in the queue to be merged into the development branch,
+a green build is required. Depending on the state of the queue or how many
+branch there is, the green build can be located at different places (refer
+to the scenarios for more information).
+
+Multiple PRs can be stacked into the queue and
+Bert-E will proceed in merging them either partially or as a whole package
+depending on the state of it.
+
+Following here a couple of scenarios of
+queue state and the behavior we can expect from Bert-E.
+
+### Scenario 1
+
+| PR/Branches |        q/1.0       |        q/2.0       |        q/3.0       |
+|:-----------:|:------------------:|:------------------:|:------------------:|
+|     #42     | :heavy_check_mark: | :heavy_check_mark: |         :x:        |
+|     #43     | :heavy_check_mark: | :heavy_check_mark: |         :x:        |
+|     #44     | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
+
+All PRs will be merged as #44 is the latest PR to enter the queue,
+it contains the code from both #42 and #43 and
+all the builds are green.
+
```

Comment:
```suggestion
At this point, we don't really care why some builds failed on `q/3.0`. It could be a flaky build (because of the code or because of the env) or even an incompatibility of the merging code with the destination branch. As long as we have a green build, we can ignore what's above.

```